Automattic/wp-calypso
# Social Previews | Add Post and Link preview for Twitter and Linkedin

## Proposed Changes

This PR adds the link and post preview for Twitter and LinkedIn which previously were in Jetpack. This is to make the component API consistent.

## Testing Instructions

- Make sure you have a self-hosted site connected to Jetpack, with a subscription that includes Social.
- Spin up Calypso, by running this branch locally or using a live link below.
- Ensure that you have Social Media accounts connected for a self-hosted WP Site
- Go to `/posts/:site` for a self-hosted WP site or a JN site
- Click on the 3 dots for any post and Click "Share"
- Click on "Preview"
- Confirm that the Facebook, Tumblr and Google preview look as before.
- Confirm that Twitter and LinkedIn preview now shows both the "Your post" and "Link preview" sections


| BEFORE | AFTER |
|--------|--------|
| <img width="1297" alt="Screenshot 2023-05-11 at 12 58 36 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/d86c7a1c-eea3-42c2-a191-b7328917bbdb"> | <img width="1212" alt="Screenshot 2023-05-11 at 12 59 10 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/a4d0bdba-62c9-4fa5-8181-7dad588675c8"> |
| <img width="1290" alt="Screenshot 2023-05-11 at 12 59 30 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/d43d73f2-8ece-4437-af8d-130187a28779"> | <img width="1217" alt="Screenshot 2023-05-11 at 12 59 49 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/2ae42500-4b6b-406f-ab2b-e76b2cb8559b"> |

## Pre-merge Checklist
